### PR TITLE
node-installer: configure and run tardev-snapshotter

### PR DESCRIPTION
**Pull request by katexochen**
With this change, we improve the stability of Contrast by deploying our own tardev-snapshotter on the host as part of the node-installer deployment. Previously, the Contrast runtime would use the tardev-snapshotter installed on the host, which is managed by Azure. However, the snapshotter is highly coupled with the guest image, and breaking changes to the snaphotter led to workload pods not being able to start in Contrast. Therefore, we are shipping a pinned snapshotter as part of the runtime. This will also allow running multiple Contrast runtime next to each other that require different versions of the snapshotter.

### CODE REVIEW

**Review by Freax13 — COMMENTED** [review]
One of the commit messages has a typo:

```diff
-  microsoft.genpolicy: don't check laryer-src-prefix 
+  microsoft.genpolicy: don't check layer-src-prefix 
```

File: node-installer/node-installer.go [hunk 1]
```diff
@@ -199,19 +199,23 @@ func patchContainerdConfig(runtimeName, basePath, configPath string, platform pl
 		existing = constants.ContainerdBaseConfig()
 	}
 
+	snapshotterName := "tardev"
```
Comment (Freax13) [inline node-installer/node-installer.go:202]: In the function below you use `"no-snapshotter"` when no snapshotter is needed? Should we also use this here?

**Review by Freax13 — APPROVED** [review]

File: node-installer/node-installer.go [hunk 2]
```diff
@@ -199,19 +199,23 @@ func patchContainerdConfig(runtimeName, basePath, configPath string, platform pl
 		existing = constants.ContainerdBaseConfig()
 	}
 
+	snapshotterName := "no-snapshotter"
```
Comment (burgerdev) [inline node-installer/node-installer.go:202]: Why is this not `tardev-$hash`?

Comment (katexochen) [reply node-installer/node-installer.go:202]: It is set to that a few lines below in case we are on a platform that uses a snapshotter.

Comment (burgerdev) [reply node-installer/node-installer.go:202]: Ah, I see - thanks.

**Review by burgerdev — APPROVED** [review]

File: node-installer/node-installer.go (hunk 2 above)
Comment (burgerdev) [reply node-installer/node-installer.go:202]: Interestingly, we're already switching on the platform outside these functions: https://github.com/edgelesssys/contrast/blob/59826da4deb184be42231a0b0bbf52634080a8db/node-installer/node-installer.go#L128-L141.

Comment (katexochen) [reply node-installer/node-installer.go:202]: That's indeed confusing (but necessary with the current structure, as some platforms won't call patchContainerdConfig.

**Pull request merged**